src/main.rs

```diff
@@ -651,6 +691,219 @@ fn should_initialize_git_repo(mut path: &Path) -> bool {
     true
 }
 
+/// Return all installed rustup toolchains that support the given `target`
+/// and meet the given MSRV.
+fn filter_toolchains_for(target: &str, msrv: &check::Version) -> Result<Vec<String>> {
```

Comment:
This function is quite complex, maybe we could add some tests to verify that it works as expected